modules/core/src/Auth/Source/AbstractSourceSelector.php

```diff
@@ -58,14 +58,17 @@ public function __construct(array $info, array $config)
      */
     public function authenticate(array &$state): void
     {
-        $source = $this->selectAuthSource();
-        $as = Auth\Source::getById($source);
-
-        if ($as === null || !in_array($source, $this->validSources, true)) {
-            throw new Exception('Invalid authentication source: ' . $source);
+        $source = $this->selectAuthSource($state);
+        if (count($source) === 1) {
+            $as = Auth\Source::getById($source[0]);
+            if ($as === null || !in_array($source[0], $this->validSources, true)) {
+                throw new Exception('Invalid authentication source: ' . $source[0]);
+            }
+
+            static::doAuthentication($as, $state);
+            return;
         }
-
-        static::doAuthentication($as, $state);
+        throw new Exception('Not implemented.');
```

Comment:
Why are you going to the changes of making this return an array now if it's not implemented? Can you then just not leave it at the single value it was?

Reply:
It was preparation for supporting the other comparisons.. Will revert